### PR TITLE
Move timeline calculations to back end, add subtasks

### PR DESCRIPTION
**Pull request by dhaselhan**
* Move all timeline logic to backend, clean up front-end, add unit tests
* Load and add subtasks to events

### CODE REVIEW

File: services/apps/alcs/src/alcs/application/application-timeline/application-timeline.service.ts [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+          },
+        },
+      },
+      order: {
+        submittedDate: 'DESC',
+      },
+    });
+
+    for (const [index, reconsideration] of reconsiderations.entries()) {
+      if (reconsideration.type.code === '33.1') {
```
Comment (mhuseinov) [inline services/apps/alcs/src/alcs/application/application-timeline/application-timeline.service.ts:222]: there is an enum for reconsideration types: RECONSIDERATION_TYPE

**Review by mhuseinov — APPROVED** [review]

File: services/apps/alcs/src/alcs/application/application-timeline/application-timeline.service.ts (hunk 1 above)
Comment (dhaselhan) [reply services/apps/alcs/src/alcs/application/application-timeline/application-timeline.service.ts:222]: switched to enum!

File: services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-timeline/notice-of-intent-timeline.service.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { CardSubtask } from '../../card/card-subtask/card-subtask.entity';
+import { NoticeOfIntentDecision } from '../../notice-of-intent-decision/notice-of-intent-decision.entity';
+import { NoticeOfIntentModification } from '../../notice-of-intent-decision/notice-of-intent-modification/notice-of-intent-modification.entity';
+import { NoticeOfIntentMeetingService } from '../notice-of-intent-meeting/notice-of-intent-meeting.service';
+import { NoticeOfIntent } from '../notice-of-intent.entity';
+import { NoticeOfIntentService } from '../notice-of-intent.service';
+
+export interface TimelineEvent {
```
[the remark this answers is not in the document]
Comment (dhaselhan) [reply services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-timeline/notice-of-intent-timeline.service.ts:11]: Done!

File: alcs-frontend/src/app/features/application/overview/overview.component.ts [hunk 3]
```diff
@@ -127,6 +53,7 @@ export class OverviewComponent implements OnInit, OnDestroy {
         title: 'Cancel Application',
       })
       .subscribe(async (didConfirm) => {
+        debugger;
```
[the remark this answers is not in the document]
Comment (dhaselhan) [reply alcs-frontend/src/app/features/application/overview/overview.component.ts:56]: Removed

**Pull request merged**